bson/CMakeLists.txt

```diff
@@ -1,18 +1,5 @@
 add_library(bson)
 
-# set boost options 
```

Comment:
in previous code logic, only bson module depends on boost, however in future multiple modules will relay on boost so we move it from bson's cmake file to project common cmake file